### PR TITLE
[Debt] Refactors Search Request table to use `ResponsiveTable`

### LINKED ISSUE #7694: ♻️ Refactor Search Requests table to use new table component
## ♻️ Debt/Refactor

We want to use new table component instead of old ApiManagedTable.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## 🌎 Localization

n/a

## ✅ Acceptance Criteria

- [ ] SearchRequestsTableApi.tsx uses new table component, does not reference ApiManagedTable, but functionality remains the same

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #7694.

## 👋 Introduction

This PR refactors Search Request table to use `ResponsiveTable`. This also adds a few things like this missing column: [Add additionalComments column](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8365/commits/4bd156939a3a0b7e3f57e0889bd7f423fe4cca3f).

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Navigate to `/admin/talent-requests`
2. Observe new table

### CODE REVIEW

File: apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      header: intl.formatMessage(adminMessages.notes),
+      cell: ({ row: { original: searchRequest } }) =>
+        notesAccessor(searchRequest, intl),
+    }),
+    columnHelper.accessor("additionalComments", {
+      id: "additionalComments",
+      enableSorting: false,
+      header: intl.formatMessage(adminMessages.details),
+      cell: ({ row: { original: searchRequest } }) =>
+        detailsAccessor(searchRequest, intl),
```
Comment (yonikid15) [inline apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx:252]: Forgot to change the name to `detailsCell`

File: apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+        }),
+        enableColumnFilter: false,
+      },
+    ),
+    columnHelper.accessor("adminNotes", {
+      id: "adminNotes",
+      enableSorting: false,
+      header: intl.formatMessage(adminMessages.notes),
+      cell: ({ row: { original: searchRequest } }) =>
+        notesAccessor(searchRequest, intl),
```
Comment (yonikid15) [inline apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx:245]: This should be named `notesAccessor`

File: apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+      id: "status",
+      header: intl.formatMessage({
+        defaultMessage: "Status",
+        id: "t3sEc+",
+        description:
+          "Title displayed on the search request table status column.",
+      }),
+      enableColumnFilter: false,
+      cell: ({ row: { original: searchRequest } }) =>
+        statusAccessor(searchRequest.status, intl),
```
Comment (yonikid15) [inline apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx:225]: This should be named `statusCell`

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Looking good so far 👍 Just some name changes that need to be made

- Pagination works
- Searching works
- Sorting works
- Hidden columns are working
- Returning back to page while maintaining search/filter/sort/hidden columns params is working

File: apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx (hunk 2 above)
Comment (mnigh) [reply apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx]: I think you meant `notesCell`, right?

Comment (yonikid15) [reply apps/web/src/components/SearchRequestTable/SearchRequestTable.tsx]: Yup 😆

**Review by yonikid15 — COMMENTED** [review]
Last week I was seeing that flash of empty table, were you or @esizer able to fix that.
I found that adding a `!isLoading` check to `hasNoVisibleRows` prevents the flash from happening. However, it doesn't address the failing query that causes it in the first place.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/7916-finalize-assessment-plan-page/apps/web/src/components/Table/ResponsiveTable/ResponsiveTable.tsx#L282-L283

**Comment by brindasasi** [comment]
In the mobile view some cards have Group and Level , Steams fields on the same line and some others in next line. 
Is it possible to have them consistent? Like always in next line ? 
Screenshot 2023-11-07 at 10 55 54 AM

**Comment by mnigh** [comment]
> In the mobile view some cards have Group and Level , Steams fields on the same line and some others in next line. Is it possible to have them consistent? Like always in next line ?

Updated in [08615d8](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8365/commits/08615d82389dd630bd56576440408c9621e0c7a1).

## Screenshot
Screen Shot 2023-11-07 at 12 41 39

**Comment by mnigh** [comment]
> Last week I was seeing that flash of empty table, were you or @esizer able to fix that. I found that adding a `!isLoading` check to `hasNoVisibleRows` prevents the flash from happening. However, it doesn't address the failing query that causes it in the first place.
> 
> https://github.com/GCTC-NTGC/gc-digital-talent/blob/7916-finalize-assessment-plan-page/apps/web/src/components/Table/ResponsiveTable/ResponsiveTable.tsx#L282-L283

Updated in [4057ff1](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8365/commits/4057ff1e07725dffaa0c9b45b18578f3f977103e). Also, made a few more small changes since your last review.

**Review by yonikid15 — APPROVED** [review]
Excellent work 🎉 looks good on my end

**Review by brindasasi — APPROVED** [review]
LGTM! 🥳

**Pull request merged**